include/glow/Base/Tensor.h

```diff
@@ -338,6 +339,23 @@ class Tensor final {
     return *this;
   }
 
+  // Dump tensor information
+  void dump(llvm::raw_ostream &os) const;
+  void dump() const;
+  std::string toString() const;
+  friend llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const Tensor &t) {
+    t.dump(os);
+    return os;
+  }
+  friend llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
+                                       const Tensor *const &t) {
```

Comment:
Hmm, reference to a pointer? Why not pass just the pointer (i.e. `const Tensor *const t`)?

Reply:
> Hmm, reference to a pointer? Why not pass just the pointer (i.e. `const Tensor *const t`)?

I saw similar function under the Type.cpp using reference to a pointer of Type . Should I keep this or change them all?

Reply:
Interesting...let me ask around. I am curious as well.

Reply:
Good catch! No, I think it is a copy&paste mistake. There is no reason to pass a const pointer by reference.

I believe it was initially written using this style because in many cases the object to be printed is passed by reference to avoid copying.

Reply:
OK. I will just simply use a pointer instead of a reference to a pointer